R/clusterSystems.R

```diff
@@ -0,0 +1,146 @@
+#' @param dataSources Optional list of input data appropriate to be used in conjunction with \code{customDistanceFn}. The entries of this list should be in the same order as \code{customDistanceFn} such that the ith element of \code{customDistanceFn} uses the data provided in the ith element of \code{dataSources} as input. Default is NULL.
+#' @param combineMatrices String indicating the method by which to combine distance matrices if more than one choice of distance data is provided. If only one type of distance data is provided, then this parameter is ignored. One of \code{c('sum', 'product', 'minimum', 'maximum')}.
+#' @param printVennDiagrams Logical flag; indicated whether or not Venn diagrams representing the set overlap of the output should be printed when the function is called. Default is TRUE.
+#'
+#' @return A named list of length 4. Elements of the list include:
+#'
+#' \code{$Clusters} A named vector of integers from 1 to k where k is the number of clusters. Names of the elements represent the genes belonging to that cluster.
+#'
+#' \code{$Best_matches} A named vector of length k representing the cluster with the best set overlap for each input system.
+#'
+#' \code{$Jaccard_indexes} The Jaccard indexes measuring the similarity between each system and the cluster which is its best match.
+#'
+#' \code{$P_values} P values representing the probability of having a Jaccard index greater than those observed by choosing a random cluster of the same size from the set of input genes.
+#'
+#' @author \href{https://orcid.org/0000-0001-5724-2252}{Rachel Silverstein} (aut)
+#'
+#' @seealso For examples of distance functions that can be used to create custom distance matrices, see \code{\link{expr_dist}} \code{\link{jaccard_dist}} \code{\link{tf_dist}}.
+#'
+#' @examples
+#' systemDB_name <- "SysDB"
+#' myDB <- fetchData(systemDB_name)
+#' rootSysIDs <- SyDBgetRootSysIDs(myDB)
+#' sys_names <- names(rootSysIDs)
+#' systems <- SyDBgetSysSymbols(myDB, sys_names)
+#'
+#' # Cluster all of the systems in the database according to
+#' # the sum of transcription factor distance and Jaccard network distance:
+#' clusterSystems(systems,
+#'                distances = c("transcription_factor", "network_jaccard"),
+#'                combineMatrices = 'sum')
+#'
+#' # Cluster all of the systems using expression profile distance
+#' # but format it like a custom distance function
+#' GEO <- fetchData("GEOprofiles")
+#' clusterSystems(systems,
+#'                distances = NULL,
+#'                customDistanceFn = list(expr_dist),
+#'                dataSources = list(GEO),
+#'                combineMatrices = NULL)
+#'
+#'
+#'
+#' @import cluster
+#' @import VennDiagram
+#' @import igraph
+#'
+#' @export
+
+
+clusterSystems <- function(systems,
+                           distances,
+                           customDistanceFn = NULL,
+                           dataSources = NULL,
+                           combineMatrices,
```

Comment:
You should allow other values of k, for exploration. You can set `length(system)` as the default, with `if (missing(k)) { k <- length(system) }